### PR TITLE
Send emails to repo owners for deploy alerts

### PR DESCRIPTION
**Pull request by bliof**
📫  Send emails to repo owners when there is a deploy alert

You have to provide DEPLOY_ALERT_EMAIL env variable in production.

By default in production emails are send with sendmail. Mailgun is also possible.
📔  When using mailgun test account you have to specify who will receive the emails in the mailgun admin panel.

In dev mode there is `mailcatcher`.

TODO:
refactor the code in https://github.com/FundingCircle/shipment_tracker/pull/201

### CODE REVIEW

**Comment by bliof** [comment]
@popovm 👀

File: app/mailers/deploy_alert_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+class DeployAlertMailer < ApplicationMailer
+  # rubocop:disable Metrics/ParameterLists
+  def deploy_alert_email(repo_owners:, repo:, region:, deployer:, deployed_at:, alert:, releases_url:)
```
Comment (popovm) [inline app/mailers/deploy_alert_mailer.rb]: Could we reduce the number of args of the method?

File: spec/jobs/deploy_alert_job_spec.rb [hunk 2]
```diff
@@ -46,5 +50,30 @@
         ),
       )
     end
+
+    it 'will notificaty the repo owners by email if there is an error' do
```
Comment (popovm) [inline spec/jobs/deploy_alert_job_spec.rb]: A small typo here

**Comment by bliof** [comment]
This code will be refactored a bit in https://github.com/FundingCircle/shipment_tracker/pull/201

File: app/mailers/deploy_alert_mailer.rb (hunk 1 above)
Comment (bliof) [inline app/mailers/deploy_alert_mailer.rb]: Not really, they are all required

File: app/jobs/deploy_alert_job.rb [hunk 3]
```diff
@@ -14,12 +16,28 @@ def perform(deploy_attrs)
...
+        DeployAlertMailer.deploy_alert_email(
+          repo_owners: repo_owners,
+          repo: app_name,
+          region: current_deploy['region'],
+          deployer: current_deploy['deployed_by'],
+          deployed_at: current_deploy['deployed_at'],
+          alert: message,
+          releases_url: releases_url,
+        ).deliver_now
+      end
```
Comment (sgerrand) [inline app/jobs/deploy_alert_job.rb]: A functional alternative:
```ruby
Repositories::RepoOwnershipRepository.new.owners_of(app_name).tap do |repo_owners|
  if repo_owners.present?
    DeployAlertMailer.deploy_alert_email(
      repo_owners: repo_owners,
      repo: app_name,
      region: current_deploy['region'],
      deployer: current_deploy['deployed_by'],
      deployed_at: current_deploy['deployed_at'],
      alert: message,
      releases_url: releases_url,
    ).deliver_now
  end
end
```
... which could then be extracted into its own method.

Comment (bliof) [inline app/jobs/deploy_alert_job.rb]: I'll extract it in a class -> e.g. https://github.com/FundingCircle/shipment_tracker/blob/e6c86dc485b97cee7fe281f012ddfd042655626c/app/jobs/deploy_alert_job.rb

File: app/models/snapshots/repo_ownership.rb [hunk 4]
```diff
@@ -5,7 +5,8 @@ module Snapshots
   class RepoOwnership < ActiveRecord::Base
     class << self
       def for(repo)
-        find_or_initialize_by(app_name: repo.name)
+        app_name = repo.is_a?(String) ? repo : repo.name
```
Comment (sgerrand) [inline app/models/snapshots/repo_ownership.rb]: 😿  It's good to avoid type checking in Ruby, where possible.

**Pull request merged**